### PR TITLE
feat(WIP): improve entity creation process #70

### PR DESCRIPTION
**Pull request by bobeal**

### CODE REVIEW

File: entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt [hunk 1]
```diff
@@ -130,7 +114,7 @@ class EntityService(
         entity: Entity,
         propertyKey: String,
         propertyValues: Map<String, List<Any>>
-    ): Property {
+    ): String {
         val propertyValue = getPropertyValueFromMap(propertyValues, NGSILD_PROPERTY_VALUE)
```
Comment (vraybaud) [inline entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:118]: Not related to improvement but readibility, maybe we could move the building part of the Property/Relationship in a builder ?

Comment (vraybaud) [reply entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:118]: Or even at least in constructors or an Util class

File: entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt [hunk 2]
```diff
@@ -130,7 +114,7 @@ class EntityService(
         entity: Entity,
         propertyKey: String,
         propertyValues: Map<String, List<Any>>
```
Comment (vraybaud) [inline entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:116]: Could we have an ExpandedProperty or ExpandedAttribute to have the same utils method as for ExpandedEntity and get relationships/properties directly ?

File: entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt [hunk 3]
```diff
@@ -24,30 +23,38 @@ class Neo4jRepository(
...
+        val query = """
+            MATCH (subject { id: '$subjectId' }), (target:Entity { id: '$targetId' })
+            CREATE (subject)-[:HAS_OBJECT]->(r:Attribute:Relationship:`${relationship.type[0]}` ${'$'}props)-[:HAS_OBJECT]->(target)
+            CREATE (r)-[:$relationshipType]->(target)
+            RETURN r.id as id
+        """
+
+        return session.query(query, mapOf("props" to relationship.nodeProperties()))
+            .toList()
+            .map { it["id"] as String }
```
Comment (vraybaud) [inline entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt:51]: Better to get first() then to get ["id"] as String or i'm wrong?

Comment (bobeal) [reply entity-service/src/main/kotlin/com/egm/stellio/entity/repository/Neo4jRepository.kt:51]: Yes indeed

File: entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt (hunk 2 above)
Comment (bobeal) [reply entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:116]: Yes, having this kind of abstraction for properties and relationships is planned for a near future

File: entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt (hunk 1 above)
Comment (bobeal) [reply entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt:118]: Also

**Pull request closed without merging**